lib/facade_converter.ts

```diff
@@ -354,6 +396,48 @@ export class FacadeConverter extends base.TranspilerBase {
     },
   };
 
+  private es6Promises: ts.Map<CallHandler> = {
+    'Promise.resolve': (c: ts.CallExpression, context: ts.Expression) => {
+      this.emit('Future.value(');
+      this.visitList(c.arguments);  // Should only be 1.
+      this.emit(')');
+    },
+    'Promise.reject': (c: ts.CallExpression, context: ts.Expression) => {
+      this.emit('Future.error(');
+      this.visitList(c.arguments);  // Should only be 1.
+      this.emit(')');
+    },
+    'Promise': (c: ts.CallExpression, context: ts.Expression) => {
+      if (c.kind != ts.SyntaxKind.NewExpression) {
+        return true;
+      }
+      this.assert(c.arguments.length == 1);
+      const callback = <ts.ArrowFunction>(c.arguments[0]);
+      this.assert(
+          callback.kind == ts.SyntaxKind.ArrowFunction ||
+          callback.kind == ts.SyntaxKind.FunctionDeclaration ||
+          callback.kind == ts.SyntaxKind.FunctionExpression);
+      this.assert(callback.parameters.length > 0 && callback.parameters.length < 3);
+
+      this.emit('(() {');  // Create a new scope.
+      this.emit('Completer _completer = new Completer();');
```

Comment:
I agree with you, but the problem is deciding where that code live. I don't mind a PromiseWrapper but we shouldn't include it anywhere. @alexeagle @mprobst What's your opinion on this? Should I create a library that's included in the pub dependencies (a kind of `ts2dart/stdlib`) or should I add a helper function in this file scope?